x-pack/plugins/spaces/public/views/nav_control/nav_control_popover.tsx

```diff
@@ -159,6 +161,20 @@ export class NavControlPopover extends Component<Props, State> {
   };
 
   private onSelectSpace = (space: Space) => {
-    this.props.spacesManager.changeSelectedSpace(space);
+    toastNotifications.add({
+      title: i18n.translate(
+        'xpack.apm.serviceDetails.enableAnomalyDetectionPanel.jobCreationFailedNotificationTitle',
+        {
+          defaultMessage: 'Switching to {spaceName} space',
+          values: {
+            spaceName: space.name,
+          },
+        }
+      ),
+      onClose: () => this.props.spacesManager.changeSelectedSpace(space),
+    });
+
+    setTimeout(() => this.props.spacesManager.changeSelectedSpace(space), 4000);
```

Comment:
Switching between spaces today already takes a decent amount of time because of the full page reload that's required, and the fact that we place users on the Kibana home screen following a space switch. So it likely takes users at least 10 seconds to switch spaces and navigate to the application they're interested in.

I feel that adding an additional 4 seconds before any of that happens would unnecessarily frustrate a large number of users. Now that the space selector menu has a descriptive aria-label ("Switch to {spaceName} space"), do we need this additional notification? We don't do this dual-notification when switching between applications, for example.

_IF_ we do need this notification, then the toast is rather hard to discover. This is perhaps more of a question for the design team, but switching a space happens in the top left of the screen, but the notification appears in the bottom right of the screen:
![image](https://user-images.githubusercontent.com/3493255/59853623-4288e000-933f-11e9-956f-b239d3d0c597.png)

Reply:
@legrego this is not going to be visible in the future, however, making toast notifications able to hidden but announced is another PR: https://github.com/elastic/eui/pull/2055

Delay you mentioned is crucial indeed.
I think you are right that we have already aria-label _before_ hit switching, so the main deal here is to confirm for a user that switching process is starting.
Would it be enough to announce "Switching..." and reload the page? How do you think?